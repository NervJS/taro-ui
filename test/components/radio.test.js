import Nerv, { findDOMNode } from 'nervjs'
import { renderToString } from 'nerv-server'
import { Simulate, renderIntoDocument } from 'nerv-test-utils'
import AtRadio from '../../.temp/components/radio/index'

const options = [
  { label: '单选项一', value: 'option1' },
  { label: '单选项二', value: 'option2', desc: '单选项描述二' },
  { label: '单选项三', value: 'option3', desc: '单选项描述三', disabled: true }
]

describe('AtRadio Snap', () => {
  it('render AtRadio', () => {
    const componet = renderToString(<AtRadio options={options} />)
    expect(componet).toMatchSnapshot()
  })
})

describe('AtRadio Event', () => {
  it('AtRadio onClick', () => {
    const onItemClick = jest.fn()
    const component = renderIntoDocument(
      <AtRadio value='option2' options={options} onClick={onItemClick} />
    )
    const items = findDOMNode(component, 'at-radio').querySelectorAll(
      '.at-radio__option'
    )
    const item0 = items[0]
    Simulate.click(item0)
    expect(onItemClick).toBeCalled()
    expect(onItemClick.mock.calls[0][0]).toBe('option1')
  })

  it('AtRadio onClick disabled, onClick not to be called', () => {
    const onItemClick = jest.fn()
    const component = renderIntoDocument(
      <AtRadio value='option2' options={options} onClick={onItemClick} />
    )
    const items = findDOMNode(component, 'at-radio').querySelectorAll(
      '.at-radio__option'
    )
    const item2 = items[2]
    Simulate.click(item2)
    expect(onItemClick).not.toBeCalled()
  })
})
