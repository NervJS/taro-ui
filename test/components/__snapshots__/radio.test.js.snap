// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtRadio Snap render AtRadio 1`] = `"<div class=\\"at-radio\\"><div class=\\"at-radio__option\\"><div class=\\"at-radio__option-wrap\\"><div class=\\"at-radio__option-container\\"><div class=\\"at-radio__title\\">单选项一</div><div class=\\"at-radio__icon\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div></div>undefined</div></div><div class=\\"at-radio__option\\"><div class=\\"at-radio__option-wrap\\"><div class=\\"at-radio__option-container\\"><div class=\\"at-radio__title\\">单选项二</div><div class=\\"at-radio__icon\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div></div><div class=\\"at-radio__desc\\">单选项描述二</div></div></div><div class=\\"at-radio__option at-radio__option--disabled\\"><div class=\\"at-radio__option-wrap\\"><div class=\\"at-radio__option-container\\"><div class=\\"at-radio__title\\">单选项三</div><div class=\\"at-radio__icon\\"><span class=\\"taro-text at-icon at-icon-check\\"></span></div></div><div class=\\"at-radio__desc\\">单选项描述三</div></div></div></div>"`;
